plugins/catalog-react/src/components/EntityRefLink/FetchedEntityRefLinks.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Entity,
+  CompoundEntityRef,
+  parseEntityRef,
+} from '@backstage/catalog-model';
+import React from 'react';
+import { EntityRefLink } from './EntityRefLink';
+import { ErrorPanel, LinkProps, Progress } from '@backstage/core-components';
+import useAsync from 'react-use/lib/useAsync';
+import { catalogApiRef } from '../../api';
+import { useApi } from '@backstage/core-plugin-api';
+
+/**
+ * Props for {@link FetchedEntityRefLinks}.
+ *
+ * @public
+ */
+export type FetchedEntityRefLinksProps<
+  TRef extends string | CompoundEntityRef | Entity,
+> = {
+  defaultKind?: string;
+  entityRefs: TRef[];
+  getTitle(entity: Entity): string | undefined;
+} & Omit<LinkProps, 'to'>;
+
+/**
+ * Shows a list of clickable links to entities with auto-fetching of entities
+ * for customising a displayed text via title attribute.
+ *
+ * @public
+ */
+export function FetchedEntityRefLinks<
+  TRef extends string | CompoundEntityRef | Entity,
+>(props: FetchedEntityRefLinksProps<TRef>) {
+  const { entityRefs, defaultKind, getTitle, ...linkProps } = props;
+
+  const catalogApi = useApi(catalogApiRef);
+
+  const {
+    value: entities = new Array<Entity>(),
+    loading,
+    error,
+  } = useAsync(async () => {
+    const refs = entityRefs.reduce((acc, current) => {
+      return 'metadata' in current ? acc : [...acc, parseEntityRef(current)];
+    }, new Array<CompoundEntityRef>());
+
+    return refs
```

Comment:
This is always gonna be truthy right? need to check `refs.length`

Reply:
Nope, this check is done when array of Entities was provided

Reply:
Hmm? In which situation is this going to be falsy, and what input would make that happen?

Reply:
'metadata' is only in Entity. So the `reduce` in the loop returns `acc`, initial value of it is empty array. So it means that it will be falsy always for that case. In case of id or ref, it will go to the right side and be `[...acc, parseEntityRef(current)]`.

Reply:
An empty array is truthy though

Reply:
my bad!

Reply:
I'll add a second test for this case